src/backend/distributed/shared_library_init.c

```diff
@@ -547,6 +548,17 @@ RegisterCitusConfigVariables(void)
 		GUC_UNIT_KB,
 		NULL, NULL, NULL);
 
+	DefineCustomIntVariable(
+		"citus.max_intermediate_result_size",
+		gettext_noop(
+			"Sets the maximum size of the intermediate results during the execution."),
+		NULL,
+		&MaxIntermediateResult,
+		1048576, -1, INT_MAX, /* max allowed size not set to MAX_KILOBYTES on purpose */
```

Comment:
which purpose?

Reply:
This part was copy-pasted from shard_max_size for the time being. I don't know the reason why they chose not to use it. I'd like to use MAX_KILOBYTES here. Do you see any problem with it?

Reply:
I don't